static/js/components/HeaderContent.jsx.template

```diff
@@ -1,92 +1,138 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { Link } from "react-router-dom";
 import { useDispatch } from "react-redux";
 
 import Box from "@mui/material/Box";
 import makeStyles from '@mui/styles/makeStyles';
 import IconButton from "@mui/material/IconButton";
 import RefreshIcon from '@mui/icons-material/Refresh';
+import SearchIcon from '@mui/icons-material/Search';
+import CloseIcon from '@mui/icons-material/Close';
 import Tooltip from "@mui/material/Tooltip";
+import { useMediaQuery } from "@mui/material";
+
 import ProfileDropdown from "./ProfileDropdown";
 import Logo from "./Logo";
 import Notifications from "./Notifications";
 import hydrate from "../actions";
 
+import QuickSearchBar from "./QuickSearchBar";
+
 const useStyles = makeStyles((theme) => ({
   topBannerContent: {
+    width: "100%",
     position: "relative",
     paddingLeft: "1rem",
-    paddingRight: "1rem",
-    paddingTop: "1.5rem",
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
   },
   title: {
-    textDecoration: "none",
+    marginLeft: "0.5rem",
     color: "white",
-    paddingLeft: "0.4rem",
-    fontSize: "200%",
+    fontSize: "160%",
     fontWeight: "bold",
     verticalAlign: "middle",
   },
   logos: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
     display: "none",
-    // Only show on bigger screens
     [theme.breakpoints.up("sm")]: {
       display: "inline-flex",
       flexDirection: "row",
     },
+    paddingRight: "0.4rem",
   },
   logoContainer: {
-    marginTop: "-0.5rem",
-    height: "50px",
+    height: "100%",
     borderRight: "1px solid white",
-    padding: "0 0.75rem",
     "&:last-of-type": {
       borderRight: "none",
       paddingRight: "0",
     },
+    "& > div": {
+      "& > img": {
+        maxHeight: "40px",
+      },
+    },
   },
 }));
+
 const HeaderContent = () => {
   const classes = useStyles();
   const dispatch = useDispatch();
 
+  const isDesktop = useMediaQuery((theme) => theme.breakpoints.up("md"));
+
   const [refreshing, setRefreshing] = useState(false);
+  const [openSearch, setOpenSearch] = useState(false);
+
   const handleRefresh = () => {
     dispatch(hydrate());
     setRefreshing(true);
     setTimeout(() => setRefreshing(false), 3000);
   };
 
+  useEffect(() => {
+    if (isDesktop) {
+      setOpenSearch(false);
+    }
+  }, [isDesktop]);
+
   return (
-    <div className={classes.topBannerContent}>
-      <div style= {%- raw -%} {{display: "inline-flex", flexDirection: "row" }} {%- endraw -%} >
+    <div className={classes.topBannerContent} style={{"{{justifyContent: openSearch ? 'flex-end' : 'space-between'}}"}}>
+      <Link onClick={() => dispatch(hydrate())} to="/" style={{"{{ display: openSearch ? 'none' : 'inline-flex', flexDirection: 'row', alignItems: 'center', textDecoration: 'none' }}"}}>
```

Comment:
Best practice may be to use logic like this to set a class, but not style, and specify the latter in CSS.

Reply:
This style is controlled by the `openSearch` variable defined in this file, so not sure how we could define it elsewhere like a global css thing.